### DISCUSSION TITLE
Multiple triggers for (same?) dialog

### DISCUSSION THREAD

**Discussion by RobHannay** [adobe/react-spectrum#6624] [Q&A]
Firstly, thanks for this incredible library. It's a joy to use, both UX and DX!

I'm initially migrating our app to use `DialogTrigger` components where appropriate.

One widget we have is a calendar timeline, where clicking on the calendar entry opens this dialog:

![image](https://github.com/adobe/react-spectrum/assets/609062/9cb41a88-a6ec-4729-83fa-c624d16e853d)

Before using RAC, clicking off the dialog would close it, but **pressing another Timeline event would open that one straight away**.

I appreciate the reasons for preventing other buttons receving presses when the dialog is modal, but what I would like to do is ideally have just one modal that is triggered by any of the events, like how it works on Google Calendar.

I'm not sure if using `DialogTrigger` is even the right approach.

I've tried using `Tooltip`s, but that doesn't feel quite right from an a11y POV.

I've tried using a combination of refs and using a single, controlled `Popover` but that caused issues as I want the `triggerRef` to reflect the event which has most recently been clicked.

I don't really mind whether the `Dialog` is the same, shared between each event (though animating its would be nice in that case!) but I would at least like to keep the existing behaviour

Existing press behaviour:

https://github.com/adobe/react-spectrum/assets/609062/5d2543e1-a0a4-48ad-8a11-96396518d858

Any wisdom would be appreciated 🙏

**Comment by snowystinger** [comment]
Thanks! I think you'll find this topic of use https://github.com/adobe/react-spectrum/issues/2061
This is working as intended at the moment and matches native behaviors with popovers on Mac. Both their calendar application and application menus.
The reason why is that everything behind the modal should actually be inert and unable to be interacted with. See the section "Notes on aria-modal and aria-hidden" https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/
